pkg/kubectl/cmd/util/editor/editoptions.go

```diff
@@ -476,6 +572,9 @@ func (o *EditOptions) visitToCreate(createVisitor resource.Visitor) error {
 }
 
 func (o *EditOptions) visitAnnotation(annotationVisitor resource.Visitor) error {
+	if o.EditMode == ApplyEditMode {
```

Comment:
I think this branching should be inside the Visit func, because this subcommand also has record flag, return  at the beginning will not record anything.